File: src/HookBridge.Domain/Entities/Notification.cs
namespace HookBridge.Domain.Entities;

public sealed class Notification : BaseEntity
{
    public string TenantId { get; set; } = string.Empty;

    public string Type { get; set; } = string.Empty;

    public string Severity { get; set; } = string.Empty;

    public string Title { get; set; } = string.Empty;

    public string Message { get; set; } = string.Empty;

    public string? ResourceType { get; set; }

    public string? ResourceId { get; set; }

    public bool IsRead { get; set; }

    public DateTime? ReadAt { get; set; }
}

File: src/HookBridge.Dashboard/src/types/notification.ts
import type { PagedRequest } from './pagination';

export type NotificationResponse = {
  id: string;
  tenantId: string;
  type: string;
  severity: 'Info' | 'Warning' | 'Error' | 'Critical' | string;
  title: string;
  message: string;
  resourceType: string | null;
  resourceId: string | null;
  isRead: boolean;
  createdAt: string;
  readAt: string | null;
};

export type NotificationSearchRequest = PagedRequest & {
  type?: string;
  severity?: string;
  isRead?: boolean;
  fromDate?: string;
  toDate?: string;
};

export type UnreadNotificationCountResponse = {
  unreadCount: number;
};
